lib/modulesync/puppet_module.rb

```diff
@@ -0,0 +1,49 @@
+require 'modulesync'
+require 'modulesync/util'
+
+module ModuleSync
+  # Provide methods to retrieve puppet module attributes
+  class PuppetModule
+    attr_reader :given_name
+    attr_reader :options
+
+    def initialize(given_name, options)
+      options ||= {}
+      @options = Util.symbolize_keys(options)
+
+      @given_name = given_name
+
+      return unless given_name.include?('/')
+
+      @repository_name = given_name.split('/').last
+      @repository_namespace = given_name.split('/')[0...-1].join('/')
+    end
+
+    def repository_name
+      @repository_name ||= given_name
+    end
+
+    def repository_namespace
+      @repository_namespace ||= @options[:namespace] || ModuleSync.options[:namespace]
+    end
+
+    def repository_path
+      @repository_path ||= "#{repository_namespace}/#{repository_name}"
+    end
+
+    def repository_remote
+      @repository_remote ||= @options[:remote] || _repository_remote
+    end
+
+    def working_directory
+      @working_directory ||= "#{ModuleSync.options[:project_root]}/#{repository_path}"
```

Comment:
```suggestion
      @working_directory ||= File.join(ModuleSync.options[:project_root], repository_path)
```

Reply:
Sorry, I'm a UNIX-only guy ;-)
But `modulesync` seems to be too... isn't it?

Do you really want this?

Reply:
While I'm also UNIX-only, what is the downside of using `File.join`? To me it also signals that first of all, it's a path. I also think the various parts (directories + file) are easier to read than a long string. To me that readability alone is why I always use it, even if I never intend to deploy on Windows.

Reply:
Good point, I followed your suggestion. Done.